src/erlazure_insights.erl

```diff
@@ -0,0 +1,130 @@
+-module(erlazure_insights).
+
+-export([
+        get_events/3,
+        get_time_offset_as_timestamp/2,
+        get_events_all/3,
+        follow_token/3,
+        create_time_bounded_filter/2
+    ]).
+
+-include("erlazure.hrl").
+
+-ifdef(TEST).
+-include_lib("eunit/include/eunit.hrl").
+-export([test_url_construction/0]).
+-endif.
+
+-type datetime() :: {{number(), number(), number},
+    {number(), number(), number()}}.
+
+-type offset() :: {seconds | minutes | hours, number()}.
+
+-spec get_events_all(string(), string(), #azure_config{}) ->
+    {ok, list()} | {{error, term()}, {accumulated, list()}}.
+get_events_all(SubscriptionId, Filter, Config=#azure_config{}) ->
+    {ok, Values, NextToken} = get_events(SubscriptionId, Filter, Config),
+    case NextToken of
+        undefined -> {ok, Values};
+        _ -> follow_token(binary_to_list(NextToken), Values, Config)
+    end.
+
+-spec get_events(string(), string(), #azure_config{}) ->
+    {ok, list(), undefined | binary()}.
+get_events(SubscriptionId, Filter, Config=#azure_config{}) ->
+    case http_request(SubscriptionId, Filter, Config) of
+        {ok, _Values, _NextToken} = Response ->
+            Response;
+        {error, _Reason} = ErrorCase ->
+            ErrorCase
+    end.
+
+-spec follow_token(string(), list(), #azure_config{}) ->
+    {ok, list()} | {{error, term()}, {accumulated, list()}}.
+follow_token(Url, DataAcc, Config) ->
+    Result = request_api(Url, Config),
+    case Result of
+        {ok, Data, undefined} ->
+            {ok, Data};
+        {ok, Data, NextToken} ->
+            follow_token(binary_to_list(NextToken), [DataAcc | Data], Config);
+        {error, _Reason} = ErrorCase ->
+            {ErrorCase, {accumulated, DataAcc}}
+    end.
+
+request_api(Url, Config) ->
+    Response = httpc:request(get, {Url, construct_headers(Config)}, [], []),
+
+    case Response of
+        {ok, {{_Version, 200, _Phrase}, _Headers, Body}} ->
+            DecodedBody = jsx:decode(list_to_binary(Body)),
+            NextLink = proplists:get_value(<<"nextLink">>, DecodedBody, undefined),
+            Values = proplists:get_value(<<"value">>, DecodedBody, undefined),
+            {ok, Values, NextLink};
+        {ok, {{_Version, Code, Phrase}, _Headers, Body}} ->
+            {error, {Code, {Phrase, Body}}};
+        {error, _Reason} = ErrorCase ->
+            ErrorCase
+    end.
+
+http_request(SubscriptionId, Filter, Config) ->
```

Comment:
current implementation forces user of the library to specify  "$filter" name each time, which seems to be strange:
`get_events(SubscriptionId, "$filter=eventTimestamp ge '<Start Time>'" , Config)`

Since this is something fixed, let's simplify it for end users to be just:
`get_events(SubscriptionId, "eventTimestamp ge '<Start Time>'" , Config)`